File: Top Interview Questions/L190. 颠倒二进制位.js
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
 var reverseBits = function (n) {
    let s=''+n
    let res=''
    for (let i = 0; i < 32; i++) {
        if (n != 0) {
            let s = Math.floor(n / 2)
            let t = n % 2
            res+=t
            n = s
        }else{
            res+=0
        }
    }
    s=res
    res=0
    for(let i=31;i>=0;i--){
        res+=(parseInt(s.charAt(i))*Math.pow(2,31-i))
    }
    return res
};

File: golang-devildyw/simple/L190颠倒二进制位.go
package main

func main() {

}

func reverseBits(n int) int {
	var result int
	for i := 0; i < 32 && n > 0; i++ {
		// n 的最后一位 与 1取并 并左移到高位 也就相当于反转嘛 再与result按位或
		result |= n & 1 << (31 - i)
		// n 向右移动 相当于舍去最后一位 上一位成为最新的最后一位
		n >>= 1
	}
	return result
}
